File: dotnet/Service/InfluxResearcherService.cs
using BoilerplateCombo.Models;
using BoilerplateCombo.Repository;
using Cassandra;
using InfluxDB.Client;
using InfluxDB.Client.Api.Domain;
using InfluxDB.Client.Writes;
using Microsoft.EntityFrameworkCore;

namespace BoilerplateCombo.Service;

public class InfluxResearcherService(InfluxDBClient client)
{
    public async Task<(List<Researcher2> researchers, int totalCount)> GetResearchersAsync(int page, int pageSize, string sortBy, bool ascending, string? filter)
    {
        string fluxQuery = $"from(bucket:\"{"mybucket"}\") |> range(start: -1h)";
        var query = await client.GetQueryApi().QueryAsync(fluxQuery, "myorg");
        foreach (var table in query)
        foreach (var record in table.Records)
            Console.WriteLine($"{record.GetTime()}: {record.GetValue()}");
        return new();
    }

    public async Task<Researcher2?> GetResearcherByIdAsync(int id)
    {
        //return await context.researcher.FindAsync(id);
        return null;
    }

    public async Task<Researcher2> AddResearcherAsync(Researcher2 researcher)
    {
        var point = PointData.Measurement("sensor_data")
            .Tag("sensor", "s1")
            .Field("temperature", 23.5)
            .Timestamp(DateTime.UtcNow, WritePrecision.Ns);
        client.GetWriteApi().WritePoint(point, "my-bucket", "my-org");
        return null;
    }

    public async Task<bool> UpdateResearcherAsync(int id, Researcher2 updatedResearcher)
    {
        // var existing = await context.researcher.FindAsync(id);
        // if (existing == null)
        //     return false;
        //
        // existing.name = updatedResearcher.name;
        // // existing.age = updatedResearcher.age;
        // await context.SaveChangesAsync();
        return true;
    }

    public async Task<bool> DeleteResearcherAsync(int id)
    {
        // var existing = await context.researcher.FindAsync(id);
        // if (existing == null)
        //     return false;
        //
        // context.researcher.Remove(existing);
        // await context.SaveChangesAsync();
        return true;
    }

    public async Task<string> ExportResearchersAsCsvAsync()
    {
        // var researchers = await context.researcher.ToListAsync();
        // var csv = "Id,CreatedAt,Name\n";
        // foreach (var researcher in researchers)
        // {
        //     csv += $"{researcher.id},{researcher.created_at:O},{EscapeCsv(researcher.name)}\n";
        //     // $",{researcher.age}\n";
        // }
        // return 
        return "";
    }

    private string EscapeCsv(string value)
    {
        if (value.Contains(",") || value.Contains("\"") || value.Contains("\n"))
        {
            value = value.Replace("\"", "\"\"");
            return $"\"{value}\"";
        }
        return value;
    }
}

File: springboot/src/main/java/com/boilerplatecombo/service/InfluxResearcherService.java
package com.boilerplatecombo.service;

import com.influxdb.client.InfluxDBClient;
import com.influxdb.client.QueryApi;
import com.influxdb.client.domain.WritePrecision;
import com.influxdb.client.write.Point;
import org.springframework.stereotype.Service;

import java.time.Instant;

@Service
public class InfluxResearcherService {

    private InfluxDBClient client;

    public InfluxResearcherService(InfluxDBClient client) {
        this.client = client;
    }

    public void queryResearchers() {
        String fluxQuery = "from(bucket: \"mybucket\") |> range(start: -1w)";
        QueryApi queryApi = client.getQueryApi();
        queryApi.query(fluxQuery).forEach(fluxTable ->
                fluxTable.getRecords().forEach(record ->
                        System.out.printf("%s: %s%n", record.getTime(), record.getValue())
                )
        );
    }

    public void writeResearcher() {
        Point point = Point.measurement("sensor_data")
                .addTag("sensor", "s1")
                .addField("temperature", 23.5)
                .time(Instant.now(), WritePrecision.NS);
        client.getWriteApiBlocking().writePoint(point);
    }
}
